infinispan-remote/src/main/java/org/hibernate/ogm/datastore/infinispanremote/query/parsing/impl/InfinispanRemoteQueryRendererDelegate.java

```diff
@@ -0,0 +1,175 @@
+
+/**
+ * Parser delegate which creates Infinispan Remote queries in form of {@link StringBuilder}s.
+ *
+ * @author Fabio Massimo Ercoli
+ */
+public class InfinispanRemoteQueryRendererDelegate extends SingleEntityQueryRendererDelegateKeepParameterPlaceholders<StringBuilder, InfinispanRemoteQueryParsingResult> {
+
+	private static final Log LOG = LoggerFactory.make( MethodHandles.lookup() );
+
+	private final SessionFactoryImplementor sessionFactory;
+	private StringBuilder sortClause;
+
+	public InfinispanRemoteQueryRendererDelegate(SessionFactoryImplementor sessionFactory, EntityNamesResolver entityNames,
+			InfinispanRemotePropertyHelper propertyHelper, Map<String, Object> namedParameters) {
+		super(
+				propertyHelper, entityNames, getSingleEntityQueryBuilder( sessionFactory, propertyHelper ),
+				namedParameters
+		);
+		this.sessionFactory = sessionFactory;
+	}
+
+	private static SingleEntityQueryBuilder<StringBuilder> getSingleEntityQueryBuilder(SessionFactoryImplementor sessionFactory, InfinispanRemotePropertyHelper propertyHelper) {
+		return SingleEntityQueryBuilder.getInstance( new InfinispanRemotePredicateFactory( sessionFactory, propertyHelper ), propertyHelper );
+	}
+
+	@Override
+	public InfinispanRemoteQueryParsingResult getResult() {
+		OgmEntityPersister ogmEntityPersister = (OgmEntityPersister) ( sessionFactory )
+				.getMetamodel()
+				.entityPersister( targetTypeName );
+
+		String table = ogmEntityPersister
+				.getEntityKeyMetadata()
+				.getTable();
+
+		StringBuilder queryBuilder;
+
+		if ( projections.isEmpty() ) {
+			queryBuilder = builder.build();
+		}
+		else {
+			queryBuilder = createFromProjection();
+			queryBuilder.append( builder.build() );
+		}
+
+		applyInheritanceStrategy( ogmEntityPersister, queryBuilder );
+
+		if ( sortClause != null ) {
+			queryBuilder.append( sortClause );
+		}
+
+		return new InfinispanRemoteQueryParsingResult( queryBuilder.toString(), table, projections );
+	}
+
+	private void applyInheritanceStrategy(OgmEntityPersister entityPersister, StringBuilder queryBuilder) {
+		String discriminatorColumnName = entityPersister.getDiscriminatorColumnName();
+
+		if ( discriminatorColumnName != null ) {
+			addConditionOnDiscriminatorValue( entityPersister, queryBuilder, discriminatorColumnName );
+		}
+		else if ( entityPersister.hasSubclasses() ) {
+			Set<String> subclassEntityNames = entityPersister.getEntityMetamodel().getSubclassEntityNames();
+			throw LOG.queriesOnPolymorphicEntitiesAreNotSupportedWithTablePerClass( "Infinispan Server", subclassEntityNames );
+		}
+	}
+
+	private void addConditionOnDiscriminatorValue(OgmEntityPersister entityPersister, StringBuilder queryBuilder, String discriminatorColumnName) {
+		Object discriminatorValue = entityPersister.getDiscriminatorValue();
+		Set<String> subclassEntityNames = entityPersister.getEntityMetamodel().getSubclassEntityNames();
+
+		boolean hasWhere = queryBuilder.toString().contains( "where" );
```

Comment:
You cannot use this approach (I think), what if one of the column in the projection is called `where_are_you`. 
Basically, there could be a `where` hidden somewhere. 
Parsing a String this way is error prone and there is always a case where it won't work.
A better approach is to keep track of the projection and conditions separately and then merge them at the end.
See what I've done in Neo4jQueryRendererDelegate.